flowey/flowey_core/src/node.rs

```diff
@@ -277,11 +287,27 @@ pub trait ClaimVar {
     fn claim(self, ctx: &mut StepCtx<'_>) -> Self::Claimed;
 }
 
+/// Read the value of one or more flowey Vars.
+///
+/// By having this be a trait, it is possible to `read` both single
+/// instances of `ReadVar` / `WriteVar`, as well as whole _collections_ of
+/// Vars.
+pub trait ReadVarValue {
+    /// The read value of Self.
+    type Value;
+    /// Read the value of the Var at runtime.
+    fn read_value(self, rt: &mut RustRuntimeServices<'_>) -> Self::Value;
```

Comment:
meta: one idea I had to streamline the whole `let foo = rt.read(foo)` dance was to play some games with registering a "global" `RustRuntimeServices` object before invoking the `|rt|` closure, and then having inherent methods on `ReadVar<T, Claimed>` to just do `foo.get()`, relying on the registered object behind the scenes. Heck - one could even use a `Deref` impl for that.

this should work fine, given that flowey steps don't run concurrently